slimdiffy/autodiff.py

```diff
@@ -1226,9 +1247,15 @@ def __call__(self, lambda_expr: Lambda) -> Lambda:
 
             # Collect derivatives for this output
             def to_grad(idx):
-                return self.derivatives[idx].idx
+                if isinstance(idx, pt.Node):
+                    idx = idx.leaf_value
+                if idx not in self.derivatives:
+                    return _ensure_tracer(np.zeros(lambda_expr.equations[idx].shape + result_eq.shape), self.supervisor)
```

Comment:
hmm this shouldn't happen? Also I think this is the wrong shape AND it assumes result_eq is in scope which it isn't here. I think the fact that an undefined variable was not encountered in testing indicates to me that this is dead code?